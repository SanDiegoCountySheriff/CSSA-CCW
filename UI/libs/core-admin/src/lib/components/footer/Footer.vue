<template>
  <v-footer
    padless
    fixed
    color="primary"
  >
    <v-col
      class="lighten-2 py-4 text-center white--text"
      cols="12"
    >
      <strong>
        {{ $t('Â© CSSA') }}
      </strong>
    </v-col>
  </v-footer>
</template>
