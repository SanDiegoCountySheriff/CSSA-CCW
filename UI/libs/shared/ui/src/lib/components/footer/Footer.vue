<template>
  <v-footer
    padless
    fixed
  >
    <v-col
      class="lighten-2 ml-4 text-center"
      style="color: #667085"
      cols="12"
    >
      © {{ new Date().getFullYear() }} — CSSA
      {{ !isDEV ? `—  ${getVersion}` : '' }}
    </v-col>
  </v-footer>
</template>
<script setup lang="ts">
import VERSION from '@shared-utils/version';
import { computed } from 'vue';

const getVersion = computed(() => VERSION);
const isDEV = process.env.NODE_ENV === 'development';
</script>
