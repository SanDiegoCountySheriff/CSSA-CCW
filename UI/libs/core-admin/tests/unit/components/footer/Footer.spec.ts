/* eslint-disable @typescript-eslint/ban-ts-comment */
import Footer from '@shared-ui/components/footer/Footer.vue'
import Vuetify from 'vuetify'
import { createTestingPinia } from '@pinia/testing'
import { createLocalVue, mount } from '@vue/test-utils'

const localVue = createLocalVue()
const pinia = createTestingPinia()
const tMock = {
  $t: text => text,
}

describe('Footer', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
    //@ts-ignore
    wrapper = mount(Footer, {
      localVue,
      vuetify,
      pinia,
      mocks: tMock,
    })
  })
  afterEach(() => {
    wrapper.destroy()
  })

  it('should match the snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
