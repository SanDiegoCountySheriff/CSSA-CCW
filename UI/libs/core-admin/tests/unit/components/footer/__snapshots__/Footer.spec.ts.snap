// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match the snapshot 1`] = `
<footer class="v-footer v-sheet theme--light v-footer--fixed v-footer--padless" style="left: 0px; right: 0px; bottom: 0px;">
  <div class="lighten-2 ml-4 text-center col col-12" style="color: rgb(102, 112, 133);">
    © 2023 — CSSA
    — #{TEMPLATE_VERSION}#
  </div>
</footer>
`;
